src/cpp/session/modules/build/SessionInstallRtools.cpp

```diff
@@ -89,6 +92,31 @@ Error installRtools()
    if (error)
       return error;
 
+   if (version == "4.2")
+   {
+      try
+      {
+         FilePath tmpFile;
+         FilePath::tempFilePath(tmpFile);
+         r::exec::RFunction("utils:::download.file")
```

Comment:
Should errors from this function be handled / logged? (Do we still want to attempt parsing / reading files on failure?)

Reply:
I'm not sure what kind of handling we would do -- if we fail to download the file, what would we be parsing? I still kept the overall fallback as our s3 copy in case of any failure

Reply:
I just think we should have something like:

```
Error error = r::exec::RFunction(<...>);
if (error) {
  LOG_ERROR(error);
  // use fallback url
}
```

In other words, check for and handle an error explicitly there.

Reply:
You could also put this snippet of code into a helper function, e.g.

```
std::string resolveRtoolsInstallerUrl(const std::string& url) { ... }
```

and that helper would just do what you have here (with a guard that returns the default URL for R < 4.2.0)